bundles/org.openhab.binding.hydrawise/src/main/java/org/openhab/binding/hydrawise/internal/HydrawiseCloudHandler.java

```diff
@@ -0,0 +1,246 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hydrawise.internal;
+
+import static org.openhab.binding.hydrawise.internal.HydrawiseBindingConstants.*;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.measure.quantity.Speed;
+import javax.measure.quantity.Temperature;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.ImperialUnits;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.openhab.binding.hydrawise.internal.api.HydrawiseAuthenticationException;
+import org.openhab.binding.hydrawise.internal.api.HydrawiseCloudApiClient;
+import org.openhab.binding.hydrawise.internal.api.HydrawiseCommandException;
+import org.openhab.binding.hydrawise.internal.api.HydrawiseConnectionException;
+import org.openhab.binding.hydrawise.internal.api.model.Controller;
+import org.openhab.binding.hydrawise.internal.api.model.CustomerDetailsResponse;
+import org.openhab.binding.hydrawise.internal.api.model.Forecast;
+import org.openhab.binding.hydrawise.internal.api.model.Relay;
+import org.openhab.binding.hydrawise.internal.api.model.StatusScheduleResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link HydrawiseCloudHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Dan Cunningham - Initial contribution
+ */
+@NonNullByDefault
+public class HydrawiseCloudHandler extends HydrawiseHandler {
+    private final Logger logger = LoggerFactory.getLogger(HydrawiseCloudHandler.class);
+    private HydrawiseCloudApiClient client;
+    int controllerId;
```

Comment:
Prefer to put all static final above the none static fields. Also can the field `controllerId` be private?